src/game_api/entity.cpp

```diff
@@ -33,9 +33,9 @@ using namespace std::chrono_literals;
 void Entity::teleport(float dx, float dy, bool s, float vx, float vy, bool snap)
 {
     if (overlay)
-        overlay->remove_item(uid);
-    overlay = NULL;
-    auto topmost = topmost_mount();
+        overlay->remove_item(this, false);
+
+    auto topmost = topmost_mount(); // we just detached from overlay, why do we check this?
```

Comment:
wondering if this is just leftover from something else?

Reply:
Maybe some old crap the UI used to need, maybe the remove_item wasn't there or maybe we wanted to really make sure we're definitely moving the topmost thing even though it should be detached already 🤷🏻‍♂️. These are like week 1 OL functions.

Reply:
I will leave it as is but move it to `ui_util` since it's only used for UI